convert2rhel/systeminfo.py

```diff
@@ -397,10 +413,58 @@ def corresponds_to_rhel_eus_release(self):
         For example if we detect CentOS Linux 8.4, the corresponding RHEL 8.4 is an EUS release, but if we detect
         CentOS Linux 8.5, the corresponding RHEL 8.5 is not an EUS release.
 
-        :return: Whether or not the current system has a EUS correspondent in RHEL.
+        :return: Whether or not the current system has an EUS correspondent in RHEL.
         :rtype: bool
         """
-        return "%s.%s" % (self.version.major, self.version.minor) in EUS_MINOR_VERSIONS
+        eus_release_date = EUS_MINOR_VERSIONS.get("%s.%s" % (self.version.major, self.version.minor), False)
+        if eus_release_date:
+            eus_release_date = dt.strptime(eus_release_date, "%Y-%m-%d")
+            current_datetime = dt.today()
+            if not self.has_internet_access:
+                return self.datetime_comparison_returns(current_datetime, eus_release_date)
+            elif current_datetime > eus_release_date:
+                try:
+                    latest_release_version = self.get_latest_distro_release_version()
+                except RepoqueryFailure:
+                    return self.datetime_comparison_returns(current_datetime, eus_release_date)
+
+                latest_major, latest_minor = latest_release_version.split(".")
+                if int(latest_minor) > int(self.version.minor):
+                    return "%s.%s" % (self.version.major, self.version.minor) in EUS_MINOR_VERSIONS
+                if int(latest_minor) < int(self.version.minor):
+                    pass
+                    # corner case that should not happen, need to decide what to do in case we land in this condition
+                    # where the users system on a version that hasn't been released yet
+        return False
```

Comment:
Tried to make the function a bit more cleaner to read, let me know if you think it works out the same way you intended to with the original one. 


```diff
diff --git a/convert2rhel/systeminfo.py b/convert2rhel/systeminfo.py
index 8b8b41a..7b129b4 100644
--- a/convert2rhel/systeminfo.py
+++ b/convert2rhel/systeminfo.py
@@ -416,26 +416,36 @@ class SystemInfo(object):
         :return: Whether or not the current system has an EUS correspondent in RHEL.
         :rtype: bool
         """
-        eus_release_date = EUS_MINOR_VERSIONS.get("%s.%s" % (self.version.major, self.version.minor), False)
-        if eus_release_date:
-            eus_release_date = dt.strptime(eus_release_date, "%Y-%m-%d")
-            current_datetime = dt.today()
-            if not self.has_internet_access:
-                return self.datetime_comparison_returns(current_datetime, eus_release_date)
-            elif current_datetime > eus_release_date:
-                try:
-                    latest_release_version = self.get_latest_distro_release_version()
-                except RepoqueryFailure:
-                    return self.datetime_comparison_returns(current_datetime, eus_release_date)
-
-                latest_major, latest_minor = latest_release_version.split(".")
-                if int(latest_minor) > int(self.version.minor):
-                    return "%s.%s" % (self.version.major, self.version.minor) in EUS_MINOR_VERSIONS
-                if int(latest_minor) < int(self.version.minor):
-                    pass
-                    # corner case that should not happen, need to decide what to do in case we land in this condition
-                    # where the users system on a version that hasn't been released yet
-        return False
+        eus_release_date = EUS_MINOR_VERSIONS.get("%s.%s" % (self.version.major, self.version.minor), "")
+        if not eus_release_date:
+            return False
+
+        # Turn eus_release_date into an datetime object
+        eus_release_date = dt.strptime(eus_release_date, "%Y-%m-%d")
+        current_datetime = dt.today()
+
+        # If our current date is not greater than eus_release_date, there is no
+        # need to continue with the function, as we will just return False at
+        # the end anyway.
+        if current_datetime < eus_release_date:
+            return False
+
+        # In case of no internet access, lets just skip the rest of the checks.
+        if not self.has_internet_access:
+            return self.datetime_comparison_returns(current_datetime, eus_release_date)
+        
+        try:
+            latest_release_version = self.get_latest_distro_release_version()
+        except RepoqueryFailure:
+            return self.datetime_comparison_returns(current_datetime, eus_release_date)
+
+        latest_major, latest_minor = latest_release_version.split(".")
+        if int(latest_minor) > int(self.version.minor):
+            return eus_release_date.strftime("%Y-%m-%d") in EUS_MINOR_VERSIONS
+        if int(latest_minor) < int(self.version.minor):
+            pass
+            # corner case that should not happen, need to decide what to do in case we land in this condition
+            # where the users system on a version that hasn't been released yet
 
     def datetime_comparison_returns(self, current_datetime, eus_release_date):
         if current_datetime > eus_release_date:
```

Reply:
Function looks better and accomplishes the same thing